### PR TITLE
feat!: pomelo

### PR DESCRIPTION
**Pull request by Lulalaby**
## Resources

- Blog article: https://discord.com/blog/usernames
- User support article: https://dis.gd/usernames
- Dev support article: https://dis.gd/app-usernames
- Official pull request: https://github.com/discord/discord-api-docs/pull/6130

## Notes

- Contributors: Either work on the `pomelo` branch or target the `pomelo` branch with your prs for this
- Currently waiting for additional docs from discord dev team

### CODE REVIEW

File: discord/member.py [hunk 1]
```diff
@@ -540,7 +541,7 @@ def display_name(self) -> str:
         if they have a guild specific nickname then that
         is returned instead.
         """
-        return self.nick or self.name
+        return self.nick or self.global_name if self._user.is_migrated else self.name or self.name
```
Comment (Lulalaby) [inline discord/member.py]: @Om1609 does that even makes sense 

cry

Comment (Om1609) [reply discord/member.py]: Uhh wtf?

Comment (Lulalaby) [reply discord/member.py]: i dunno how to do it better

https://github.com/Aiko-IT-Systems/DisCatSharp/blob/a1bd54efb9d09d069257d6f954cab6ba3bedbc28/DisCatSharp/Entities/Guild/DiscordMember.cs#L152-L159

**Comment by Dorukyum** [comment]
Testing required

**Comment by Lulalaby** [comment]
Nelo tested it successfully already.

Need one or two more testers for that.

Currently can only be tested if the bot shares a guild with a discord staff.

**Comment by NeloBlivion** [comment]
Just went and tested it on DCS for staff members and everything seemed to go well

**Pull request merged**